### PR TITLE
Refactor EmailPassword Auth tests.

### PR DESCRIPTION
**Pull request by cmelchior**
In the past, we have seen quite a lot of flakiness in the EmailPasswordAuth tests. The root cause was never discovered. With the refactor to Ktor 2, we started to see them again. This PR refactors the tests, so instead of modifying the config of a single app, we now use different test apps.

This seems to fix the flakiness + has the advantage of improving the speed of running tests.

### CODE REVIEW

File: test/sync/src/androidTest/kotlin/io/realm/kotlin/test/shared/SyncedRealmTests.kt [hunk 1]
```diff
@@ -258,24 +260,24 @@ class SyncedRealmTests {
...
+            val deferred = async {
+                Realm.open(config)
+                // Make sure that the test eventually fail. Coroutines can cancel a delay
+                // so this doesn't always block the test for 10 seconds.
+                delay(10 * 1000)
+                channel.trySend(AssertionError("Realm was successfully opened"))
             }
+
+            val error = channel.receive()
+            assertTrue(error is UnrecoverableSyncException, "Was $error")
```
Comment (rorbech) [inline test/sync/src/androidTest/kotlin/io/realm/kotlin/test/shared/SyncedRealmTests.kt:273]: Isn't this a job for `assertFailsWithMessage`?

File: test/sync/src/commonMain/kotlin/io/realm/kotlin/test/mongodb/util/AdminApi.kt [hunk 2]
```diff
@@ -94,14 +107,9 @@ interface AdminApi {
...
-     * Set whether or not automatic confirmation is enabled.
-     */
-    suspend fun setAutomaticConfirmation(enabled: Boolean)
-
-    /**
-     * Set whether or not custom confirmation is enabled.
+     * Set type of authentication method used for the EmailAuth provider.
      */
-    suspend fun setCustomConfirmation(enabled: Boolean)
+    suspend fun setAuth(auth: UserConfirmationMode)
```
Comment (rorbech) [inline test/sync/src/commonMain/kotlin/io/realm/kotlin/test/mongodb/util/AdminApi.kt:112]: Do we even use this anymore?

**Review by rorbech — COMMENTED** [review]
LGTM - Only very minor comments

**Review by rorbech — APPROVED** [review]
LGTM - Only very minor comments (in previous comment ... forgot to approve) ... and of course ... when CI is green

File: test/sync/src/commonMain/kotlin/io/realm/kotlin/test/mongodb/util/AdminApi.kt (hunk 2 above)
Comment (cmelchior) [reply test/sync/src/commonMain/kotlin/io/realm/kotlin/test/mongodb/util/AdminApi.kt:112]: No, I was a bit afraid of removing it, but I should just get rid of it.

**Pull request merged**